app/src/main/java/memphis/myapplication/utilities/BloomFilter.java

```diff
@@ -0,0 +1,387 @@
+        {
+            long hash = murmurHash3((int) salt_[i], key);
+            bit_index = hash % table_size_;
+            bit = bit_index % bits_per_char;
+            long value = bit_table_[(int) (bit_index/bits_per_char)] | bit_mask[(int) bit];
+            bit_table_[(int) (bit_index/bits_per_char)] = (byte) value;
+        }
+        ++inserted_element_count_;
+    }
+
+    public boolean
+    contains(String key)
+    {
+        long bit_index = 0;
+        long bit = 0;
+
+        for (int i = 0; i < salt_.length; ++i)
+        {
+            long hash = murmurHash3((int) salt_[i], key);
+            bit_index = hash % table_size_;
+            bit = bit_index % bits_per_char;
+            byte bitTableEntry = (byte) (bit_table_[(int) (bit_index/bits_per_char)] & bit_mask[(int) bit]);
+            if (toUnsignedInt(bitTableEntry) != bit_mask[(int) bit]) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    public String toString() {
+        String out = "";
+        int i = 0;
+        for (byte element : bit_table_) {
+            out += element;
+        }
+        return out;
+    }
+
+    public Name
+    appendToName(Name prefix) {
+        prefix.append(Component.fromNumber(projected_element_count_));
+        prefix.append(Component.fromNumber((int)(desired_false_positive_probability_ * 1000)));
+        prefix.append(bit_table_);
+        return prefix;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == this) {
+            return true;
+        }
+
+        if (!(o instanceof BloomFilter)) {
+            return false;
+        }
+
+        byte[] otherTable = ((BloomFilter) o).bit_table_;
+        if (otherTable.length != bit_table_.length) {
+            return false;
+        }
+
+        for (int i = 0; i < bit_table_.length; i++) {
+            if (otherTable[i] != bit_table_[i]) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static long murmurHash3(int seed, long key)
+    {
+        com.google.common.hash.HashFunction hashImplementation = Hashing.murmur3_32(seed);
+        return toUnsignedLong(hashImplementation.newHasher().putLong(key).hash().asInt());
+    }
+
+    public static long murmurHash3(int seed, int key)
+    {
+        com.google.common.hash.HashFunction hashImplementation = Hashing.murmur3_32(seed);
+        return toUnsignedLong(hashImplementation.newHasher().putInt(key).hash().asInt());
+    }
+
+    public static long murmurHash3(int seed, String key)
+    {
+        com.google.common.hash.HashFunction hashImplementation = Hashing.murmur3_32(seed);
+        return toUnsignedLong(hashImplementation.newHasher().putString(key, Charsets.UTF_8).hash().asInt());
+    }
+
+    public static long toUnsignedLong(int x) {
+        return x & 0x00000000ffffffffL;
+    }
+
+    public static int toUnsignedInt(byte x) { return x & 0xFF; }
+
+}
+class BloomParameters
```

Comment:
All of this class is c style formatted. Should we format more like Java?

Reply:
I can. What are the major differences other than, say, like breaks in the method headers?